siliconcompiler/core.py

```diff
@@ -4935,32 +4918,63 @@ def _is_builtin(self, tool, task):
         return task in self.builtin
 
     #######################################
-    def _get_steps_by_task(self, flow=None):
+    def _get_flowgraph_entry_nodes(self, flow=None):
+        '''
+        Collect all step/indecies that represent the entry
+        nodes for the flowgraph
+        '''
+        if not flow:
+            flow = self.get('option', 'flow')
+
+        nodes = []
+        for step in self.getkeys('flowgraph', flow):
+            for index in self.getkeys('flowgraph', flow, step):
+                if not self.get('flowgraph', flow, step, index, 'input'):
+                    nodes.append((step, index))
+        return nodes
+
+    #######################################
+    def _get_flowgraph_exit_nodes(self, flow=None):
         '''
-        Collect all steps and indicies and organize by tasks
-        Returns dict(task, [(step, index)])
+        Collect all step/indecies that represent the exit
+        nodes for the flowgraph
         '''
         if not flow:
             flow = self.get('option', 'flow')
 
-        tasks = {}
+        inputnodes = []
         for step in self.getkeys('flowgraph', flow):
             for index in self.getkeys('flowgraph', flow, step):
-                tasks.setdefault(self._get_task(step, index, flow=flow), []).append((step, index))
-        return tasks
+                inputnodes.extend(self.get('flowgraph', flow, step, index, 'input'))
+        nodes = []
+        for step in self.getkeys('flowgraph', flow):
+            for index in self.getkeys('flowgraph', flow, step):
+                if (step, index) not in inputnodes:
+                    nodes.append((step, index))
+        return nodes
+
+    #######################################
+    def _getcollectdir(self, jobname=None):
+        '''
+        Get absolute path to collected files directory
+        '''
+
+        return os.path.join(self._getworkdir(jobname=jobname), 'sc_collected_files')
```

Comment:
Nice clean up to use this instead of `import/0/inputs`, do you think we want a check somewhere to prevent someone from naming an actual step "sc_collected_files" (as unlikely as that might be...)

Reply:
probably should add a check just in case